### ISSUE TITLE
[Bug]: infinity::ParserException, Boolean type should be true or false.

### ISSUE THREAD

**Issue by chrysanthemum-boy** [infiniflow/infinity#842]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch name

main

### Commit ID

80d04e1

### Actual behavior

```
terminate called after throwing an instance of 'infinity::ParserException'
  what():  Boolean type should be true or false @src/parser/type/data_type.cpp:489
Aborted (core dumped)
```

### Steps to reproduce

```Markdown
Function test case can be found in test_import.py `test_csv_with_different_delimiter`
```

**Comment by Ognimalf** [comment]
Fixed by #880

**Comment by chrysanthemum-boy** [comment]
It seems fixed.
